src/circuits/etable_compact/op_configure/op_load.rs

```diff
@@ -207,6 +215,44 @@ impl<F: FieldExt> EventTableOpcodeConfigBuilder<F> for LoadConfigBuilder {
             }),
         );
 
+        constraint_builder.push(
+            "op_load value: value = padding + value_in_heap",
+            Box::new(move |meta| {
+                let padding = is_one_byte.expr(meta) * constant_from!(0xffffff00)
+                    + is_two_bytes.expr(meta) * constant_from!(0xffff0000)
+                    + (constant_from!(1) - is_eight_bytes.expr(meta)) * (vtype.expr(meta) - constant_from!(1)) * constant_from!(0xffffffff00000000);
+                vec![res.expr(meta) - value_in_heap.expr(meta) 
+                - highest_bit.expr(meta) * is_sign.expr(meta) * padding]
+            }),
+        );
+
+        constraint_builder.push(
+            "op_load highest_bit",
+            Box::new(move |meta| {
```

Comment:
I think the highest bits should be the first bits of loading value in big-endian instead of the res value.
So I guess we should pick the u4 of loading_value according to the loading size.
```
highest_u4 = is_one_byte * loading_u4[0] + is_two_byte * loading_u4[3]...
decompose(highest_u4) = [highest bit, second bit ...]
```

Reply:
cool, fixed!